File: Backend/IS.Domain/Entities/Product.cs
namespace IS.Domain.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public bool IsDeleted { get; set; } = false;
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime? UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
        public User User { get; set; } = null!;
        public ICollection<ProductSupplier> ProductSuppliers { get; set; } = new List<ProductSupplier>();
    }
}

File: Frontend/inventory-system/src/app/interfaces/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  createdAt: string;
  updatedAt: string | null;
  suppliers: ProductSupplier[];
}

export interface ProductSupplier {
  id: number;
  supplierId: number;
  supplierName: string;
  price: number;
  stock: number;
  batchNumber: string;
}

export interface CreateProduct {
  name: string;
  description: string;
  suppliers: CreateProductSupplier[];
}

export interface CreateProductSupplier {
  supplierId: number;
  price: number;
  stock: number;
  batchNumber: string;
}